x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/utils/wrap_suppressed_alerts.ts

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import objectHash from 'object-hash';
+import pick from 'lodash/pick';
+import get from 'lodash/get';
+import sortBy from 'lodash/sortBy';
+
+import type { SuppressionFieldsLatest } from '@kbn/rule-registry-plugin/common/schemas';
+import {
+  ALERT_SUPPRESSION_DOCS_COUNT,
+  ALERT_INSTANCE_ID,
+  ALERT_SUPPRESSION_TERMS,
+  ALERT_SUPPRESSION_START,
+  ALERT_SUPPRESSION_END,
+  TIMESTAMP,
+} from '@kbn/rule-data-utils';
+import type { SignalSourceHit } from '../types';
+
+import type {
+  BaseFieldsLatest,
+  WrappedFieldsLatest,
+} from '../../../../../common/api/detection_engine/model/alerts';
+import type { ConfigType } from '../../../../config';
+import type { CompleteRule, ThreatRuleParams } from '../../rule_schema';
+import type { IRuleExecutionLogForExecutors } from '../../rule_monitoring';
+import { buildBulkBody } from '../factories/utils/build_bulk_body';
+
+import type { BuildReasonMessage } from './reason_formatters';
+
+/**
+ * wraps suppressed threshold alerts
+ * first, transforms aggregation threshold buckets to hits
+ * creates instanceId hash, which is used to search suppressed on time interval alerts
+ * populates alert's suppression fields
+ */
+export const wrapSuppressedAlerts = ({
+  events,
+  spaceId,
+  completeRule,
+  mergeStrategy,
+  indicesToQuery,
+  buildReasonMessage,
+  alertTimestampOverride,
+  ruleExecutionLogger,
+  publicBaseUrl,
+  primaryTimestamp,
+  secondaryTimestamp,
+}: {
+  events: SignalSourceHit[];
+  spaceId: string;
+  completeRule: CompleteRule<ThreatRuleParams>;
```

Comment:
Wouldn't this wrapper apply to most rule types? I noticed that https://github.com/elastic/kibana/pull/174241/files#diff-1db264da3d27dbbc4e709f64e37298dd7c92635baa716f1d71c86d78f7c8d27fR36  it references `Threshold`, and the 'completeRule' is specifically of type 'ThreatRuleParams'.

Reply:
Removed all threshold mentions, leftovers from other comments

Reply:
We discussed that point and agreed to pipe the new suppression rule types for the `completeRule`